python/shotgun_model/shotgun_query_model.py

```diff
@@ -318,122 +287,117 @@ def reset(self):
             "email support@shotgunsoftware.com."
         )
 
-    ############################################################################
-    # protected properties
-
-    def _get_cache_path(self):
+    def hasChildren(self, index):
```

Comment:
not sure. a lot of the code you are commenting on is stuff that i have moved around, i was wondering the same thing as i was shifting it :) I think @josh-t wrote it that way to be resilient, which seems like a nice approach to me.